[2023-10-19 07:44:49,660][pinnsjax.utils.utils][INFO] - Enforcing tags! <cfg.extras.enforce_tags=True>
[2023-10-19 07:44:49,664][pinnsjax.train][INFO] - Instantiating time domain <pinnsjax.data.TimeDomain>
[2023-10-19 07:44:49,664][pinnsjax.train][INFO] - Instantiating spatial domain <pinnsjax.data.Interval>
[2023-10-19 07:44:49,665][pinnsjax.train][INFO] - Instantiating mesh <pinnsjax.data.Mesh>
[2023-10-19 07:44:49,668][pinnsjax.utils.utils][INFO] - Data is available.
[2023-10-19 07:44:49,670][pinnsjax.train][INFO] - Instantiating training dataset number 1: <pinnsjax.data.DiscreteMeshSampler>
[2023-10-19 07:44:49,814][jax._src.xla_bridge][INFO] - Unable to initialize backend 'rocm': NOT_FOUND: Could not find registered platform with name: "rocm". Available platform names are: CUDA
[2023-10-19 07:44:49,816][jax._src.xla_bridge][INFO] - Unable to initialize backend 'tpu': INTERNAL: Failed to open libtpu.so: libtpu.so: cannot open shared object file: No such file or directory
[2023-10-19 07:44:51,504][pinnsjax.train][INFO] - Instantiating training dataset number 2: <pinnsjax.data.DiscreteMeshSampler>
[2023-10-19 07:44:51,544][pinnsjax.train][INFO] - Instantiating validation dataset number 1: <pinnsjax.data.DiscreteMeshSampler>
[2023-10-19 07:44:51,570][pinnsjax.train][INFO] - Instantiating prediction dataset number 1: <pinnsjax.data.DiscreteMeshSampler>
[2023-10-19 07:44:51,573][pinnsjax.train][INFO] - Instantiating datamodule <pinnsjax.data.PINNDataModule>
[2023-10-19 07:44:51,575][pinnsjax.train][INFO] - Instantiating neural net <pinnsjax.models.FCN>
[2023-10-19 07:44:52,603][pinnsjax.train][INFO] - Instantiating model <pinnsjax.models.PINNModule>
[2023-10-19 07:44:52,611][pinnsjax.utils.utils][INFO] - Weights are available.
[2023-10-19 07:44:52,845][pinnsjax.train][INFO] - Instantiating trainer <pinnsjax.trainer.Trainer>
[2023-10-19 07:44:52,847][pinnsjax.train][INFO] - Starting training!
[2023-10-19 07:44:52,848][pinnsjax.trainer.trainer][INFO] - Training with Adam Optimizer
[2023-10-19 07:45:03,521][pinnsjax.utils.utils][INFO] - Enforcing tags! <cfg.extras.enforce_tags=True>
[2023-10-19 07:45:03,526][pinnsjax.train][INFO] - Instantiating time domain <pinnsjax.data.TimeDomain>
[2023-10-19 07:45:03,527][pinnsjax.train][INFO] - Instantiating spatial domain <pinnsjax.data.Interval>
[2023-10-19 07:45:03,528][pinnsjax.train][INFO] - Instantiating mesh <pinnsjax.data.Mesh>
[2023-10-19 07:45:03,531][pinnsjax.utils.utils][INFO] - Data is available.
[2023-10-19 07:45:03,534][pinnsjax.train][INFO] - Instantiating training dataset number 1: <pinnsjax.data.MeshSampler>
[2023-10-19 07:45:03,691][jax._src.xla_bridge][INFO] - Unable to initialize backend 'rocm': NOT_FOUND: Could not find registered platform with name: "rocm". Available platform names are: CUDA
[2023-10-19 07:45:03,693][jax._src.xla_bridge][INFO] - Unable to initialize backend 'tpu': INTERNAL: Failed to open libtpu.so: libtpu.so: cannot open shared object file: No such file or directory
[2023-10-19 07:45:05,717][pinnsjax.train][INFO] - Instantiating validation dataset number 1: <pinnsjax.data.MeshSampler>
[2023-10-19 07:45:05,744][pinnsjax.train][INFO] - Instantiating prediction dataset number 1: <pinnsjax.data.MeshSampler>
[2023-10-19 07:45:05,747][pinnsjax.train][INFO] - Instantiating datamodule <pinnsjax.data.PINNDataModule>
[2023-10-19 07:45:05,748][pinnsjax.train][INFO] - Instantiating neural net <pinnsjax.models.FCN>
[2023-10-19 07:45:06,793][pinnsjax.train][INFO] - Instantiating model <pinnsjax.models.PINNModule>
[2023-10-19 07:45:06,993][pinnsjax.train][INFO] - Instantiating trainer <pinnsjax.trainer.Trainer>
[2023-10-19 07:45:06,996][pinnsjax.train][INFO] - Starting training!
[2023-10-19 07:45:06,996][pinnsjax.trainer.trainer][INFO] - Training with Adam Optimizer
[2023-10-19 07:45:12,845][pinnsjax.utils.utils][INFO] - Enforcing tags! <cfg.extras.enforce_tags=True>
[2023-10-19 07:45:12,852][pinnsjax.train][INFO] - Instantiating time domain <pinnsjax.data.TimeDomain>
[2023-10-19 07:45:12,854][pinnsjax.train][INFO] - Instantiating spatial domain <pinnsjax.data.Interval>
[2023-10-19 07:45:12,856][pinnsjax.train][INFO] - Instantiating mesh <pinnsjax.data.Mesh>
[2023-10-19 07:45:12,859][pinnsjax.utils.utils][INFO] - Data is available.
[2023-10-19 07:45:12,863][pinnsjax.train][INFO] - Instantiating training dataset number 1: <pinnsjax.data.MeshSampler>
[2023-10-19 07:45:13,038][jax._src.xla_bridge][INFO] - Unable to initialize backend 'rocm': NOT_FOUND: Could not find registered platform with name: "rocm". Available platform names are: CUDA
[2023-10-19 07:45:13,040][jax._src.xla_bridge][INFO] - Unable to initialize backend 'tpu': INTERNAL: Failed to open libtpu.so: libtpu.so: cannot open shared object file: No such file or directory
[2023-10-19 07:45:14,826][pinnsjax.train][INFO] - Instantiating training dataset number 2: <pinnsjax.data.InitialCondition>
[2023-10-19 07:45:14,854][pinnsjax.train][INFO] - Instantiating training dataset number 3: <pinnsjax.data.DirichletBoundaryCondition>
[2023-10-19 07:45:14,857][pinnsjax.train][INFO] - Instantiating validation dataset number 1: <pinnsjax.data.MeshSampler>
[2023-10-19 07:45:14,883][pinnsjax.train][INFO] - Instantiating prediction dataset number 1: <pinnsjax.data.MeshSampler>
[2023-10-19 07:45:14,886][pinnsjax.train][INFO] - Instantiating datamodule <pinnsjax.data.PINNDataModule>
[2023-10-19 07:45:14,888][pinnsjax.train][INFO] - Instantiating neural net <pinnsjax.models.FCN>
[2023-10-19 07:45:15,916][pinnsjax.train][INFO] - Instantiating model <pinnsjax.models.PINNModule>
[2023-10-19 07:45:16,127][pinnsjax.train][INFO] - Instantiating trainer <pinnsjax.trainer.Trainer>
[2023-10-19 07:45:16,130][pinnsjax.train][INFO] - Starting training!
[2023-10-19 07:45:16,130][pinnsjax.trainer.trainer][INFO] - Training with Adam Optimizer
in (50, 1) (50, 1)
[2023-10-19 07:45:16,246][pinnsjax.utils.utils][ERROR] - 
jax.errors.SimplifiedTraceback: For simplicity, JAX has removed its internal frames from the traceback of the following exception. Set JAX_TRACEBACK_FILTERING=off to include these.

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/reza/tmp/pinns-jax/pinnsjax/utils/utils.py", line 72, in wrap
    metric_dict, object_dict = task_func(
  File "/home/reza/tmp/pinns-jax/pinnsjax/train.py", line 138, in train
    trainer.fit(model=model, datamodule=datamodule)
  File "/home/reza/tmp/pinns-jax/pinnsjax/trainer/trainer.py", line 109, in fit
    loss, extra_variables, model.trainable_variables, model.opt_state = model.train_step(model.trainable_variables,
  File "/home/reza/tmp/pinns-jax/pinnsjax/models/pinn_module.py", line 139, in train_step
    loss_pred, grads = jax.value_and_grad(self.model_step, has_aux=True)(params, batch)
  File "/home/reza/tmp/pinns-jax/pinnsjax/models/pinn_module.py", line 130, in model_step
    loss, preds = self.function_mapping[loss_fn_name](params, data, loss, self.functions)
  File "/home/reza/tmp/pinns-jax/pinnsjax/data/sampler/initial_condition.py", line 76, in loss_fn
    print(outputs['u'].shape, outputs['v'].shape, u['u'].shape, u['v'].shape)
KeyError: 'v'
[2023-10-19 07:45:16,247][pinnsjax.utils.utils][INFO] - Output dir: /home/reza/tmp/pinns-jax/examples/burgers_continuous_forward/outputs/07-45-12
Error executing job with overrides: ['trainer.enable_progress_bar=False']
jax.errors.SimplifiedTraceback: For simplicity, JAX has removed its internal frames from the traceback of the following exception. Set JAX_TRACEBACK_FILTERING=off to include these.

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/reza/tmp/pinns-jax/examples/burgers_continuous_forward/train.py", line 51, in main
    metric_dict, _ = pinnsjax.train(
  File "/home/reza/tmp/pinns-jax/pinnsjax/utils/utils.py", line 84, in wrap
    raise ex
  File "/home/reza/tmp/pinns-jax/pinnsjax/utils/utils.py", line 72, in wrap
    metric_dict, object_dict = task_func(
  File "/home/reza/tmp/pinns-jax/pinnsjax/train.py", line 138, in train
    trainer.fit(model=model, datamodule=datamodule)
  File "/home/reza/tmp/pinns-jax/pinnsjax/trainer/trainer.py", line 109, in fit
    loss, extra_variables, model.trainable_variables, model.opt_state = model.train_step(model.trainable_variables,
  File "/home/reza/tmp/pinns-jax/pinnsjax/models/pinn_module.py", line 139, in train_step
    loss_pred, grads = jax.value_and_grad(self.model_step, has_aux=True)(params, batch)
  File "/home/reza/tmp/pinns-jax/pinnsjax/models/pinn_module.py", line 130, in model_step
    loss, preds = self.function_mapping[loss_fn_name](params, data, loss, self.functions)
  File "/home/reza/tmp/pinns-jax/pinnsjax/data/sampler/initial_condition.py", line 76, in loss_fn
    print(outputs['u'].shape, outputs['v'].shape, u['u'].shape, u['v'].shape)
KeyError: 'v'

Set the environment variable HYDRA_FULL_ERROR=1 for a complete stack trace.
[2023-10-19 07:45:47,458][pinnsjax.train][INFO] - Elapsed time: 54.610355854034424
[2023-10-19 07:45:47,460][pinnsjax.train][INFO] - MEAN time: 0.0005667209625244141  -   0.0007537603378295898 -  0.0006976127624511719
[2023-10-19 07:45:47,460][pinnsjax.train][INFO] - Starting validation!
[2023-10-19 07:45:47,464][pinnsjax.train][INFO] - Plotting the results
[2023-10-19 07:45:48,871][pinnsjax.utils.plotting][INFO] - Image saved at /home/reza/tmp/pinns-jax/examples/burgers_discrete_inverse/outputs/07-44-49/fig
[2023-10-19 07:45:49,667][pinnsjax.utils.utils][INFO] - Output dir: /home/reza/tmp/pinns-jax/examples/burgers_discrete_inverse/outputs/07-44-49
[2023-10-19 07:45:49,667][pinnsjax.utils.utils][INFO] - Retrieved metric value! <val/error_u=0.004949159920215607>
[2023-10-19 07:45:49,668][pinnsjax.utils.utils][INFO] - Retrieved metric value! <l1=0.9973235726356506>
[2023-10-19 07:45:49,668][pinnsjax.utils.utils][INFO] - Retrieved metric value! <l2=-5.607864856719971>
[2023-10-19 07:46:02,899][pinnsjax.train][INFO] - Elapsed time: 55.90331721305847
[2023-10-19 07:46:02,901][pinnsjax.train][INFO] - MEAN time: 0.0010356903076171875  -   0.0010360479354858398 -  0.0010409355163574219
[2023-10-19 07:46:02,901][pinnsjax.train][INFO] - Starting validation!
[2023-10-19 07:46:02,904][pinnsjax.train][INFO] - Plotting the results
[2023-10-19 07:46:03,662][pinnsjax.utils.plotting][INFO] - Image saved at /home/reza/tmp/pinns-jax/examples/burgers_continuous_inverse/outputs/07-45-03/fig
[2023-10-19 07:46:04,659][pinnsjax.utils.utils][INFO] - Output dir: /home/reza/tmp/pinns-jax/examples/burgers_continuous_inverse/outputs/07-45-03
[2023-10-19 07:46:04,659][pinnsjax.utils.utils][INFO] - Retrieved metric value! <val/error_u=0.00946053583174944>
[2023-10-19 07:46:04,660][pinnsjax.utils.utils][INFO] - Retrieved metric value! <l1=0.9987875819206238>
[2023-10-19 07:46:04,660][pinnsjax.utils.utils][INFO] - Retrieved metric value! <l2=-5.664256572723389>
[2023-10-19 07:46:25,146][pinnsjax.utils.utils][INFO] - Enforcing tags! <cfg.extras.enforce_tags=True>
[2023-10-19 07:46:25,149][pinnsjax.train][INFO] - Instantiating time domain <pinnsjax.data.TimeDomain>
[2023-10-19 07:46:25,150][pinnsjax.train][INFO] - Instantiating spatial domain <pinnsjax.data.Interval>
[2023-10-19 07:46:25,151][pinnsjax.train][INFO] - Instantiating mesh <pinnsjax.data.Mesh>
[2023-10-19 07:46:25,153][pinnsjax.utils.utils][INFO] - Data is available.
[2023-10-19 07:46:25,155][pinnsjax.train][INFO] - Instantiating training dataset number 1: <pinnsjax.data.MeshSampler>
[2023-10-19 07:46:25,340][jax._src.xla_bridge][INFO] - Unable to initialize backend 'rocm': NOT_FOUND: Could not find registered platform with name: "rocm". Available platform names are: CUDA
[2023-10-19 07:46:25,342][jax._src.xla_bridge][INFO] - Unable to initialize backend 'tpu': INTERNAL: Failed to open libtpu.so: libtpu.so: cannot open shared object file: No such file or directory
[2023-10-19 07:46:27,132][pinnsjax.train][INFO] - Instantiating training dataset number 2: <pinnsjax.data.InitialCondition>
[2023-10-19 07:46:27,156][pinnsjax.train][INFO] - Instantiating training dataset number 3: <pinnsjax.data.DirichletBoundaryCondition>
[2023-10-19 07:46:27,160][pinnsjax.train][INFO] - Instantiating validation dataset number 1: <pinnsjax.data.MeshSampler>
[2023-10-19 07:46:27,199][pinnsjax.train][INFO] - Instantiating prediction dataset number 1: <pinnsjax.data.MeshSampler>
[2023-10-19 07:46:27,203][pinnsjax.train][INFO] - Instantiating datamodule <pinnsjax.data.PINNDataModule>
[2023-10-19 07:46:27,205][pinnsjax.train][INFO] - Instantiating neural net <pinnsjax.models.FCN>
[2023-10-19 07:46:28,218][pinnsjax.train][INFO] - Instantiating model <pinnsjax.models.PINNModule>
[2023-10-19 07:46:28,417][pinnsjax.train][INFO] - Instantiating trainer <pinnsjax.trainer.Trainer>
[2023-10-19 07:46:28,419][pinnsjax.train][INFO] - Starting training!
[2023-10-19 07:46:28,420][pinnsjax.trainer.trainer][INFO] - Training with Adam Optimizer
[2023-10-19 07:47:39,389][pinnsjax.train][INFO] - Elapsed time: 70.9694664478302
[2023-10-19 07:47:39,390][pinnsjax.train][INFO] - MEAN time: 0.0021610260009765625  -   0.0021584033966064453 -  0.0020771026611328125
[2023-10-19 07:47:39,391][pinnsjax.train][INFO] - Starting validation!
[2023-10-19 07:47:39,443][pinnsjax.train][INFO] - Plotting the results
[2023-10-19 07:47:40,210][pinnsjax.utils.plotting][INFO] - Image saved at /home/reza/tmp/pinns-jax/examples/burgers_continuous_forward/outputs/07-46-25/fig
[2023-10-19 07:47:41,238][pinnsjax.utils.utils][INFO] - Output dir: /home/reza/tmp/pinns-jax/examples/burgers_continuous_forward/outputs/07-46-25
[2023-10-19 07:47:41,238][pinnsjax.utils.utils][INFO] - Retrieved metric value! <val/error_u=0.018565678969025612>
[2023-10-19 07:51:44,215][pinnsjax.utils.utils][INFO] - Enforcing tags! <cfg.extras.enforce_tags=True>
[2023-10-19 07:51:44,218][pinnsjax.train][INFO] - Instantiating time domain <pinnsjax.data.TimeDomain>
[2023-10-19 07:51:44,219][pinnsjax.train][INFO] - Instantiating spatial domain <pinnsjax.data.Interval>
[2023-10-19 07:51:44,220][pinnsjax.train][INFO] - Instantiating mesh <pinnsjax.data.Mesh>
[2023-10-19 07:51:44,222][pinnsjax.utils.utils][INFO] - Data is available.
[2023-10-19 07:51:44,224][pinnsjax.train][INFO] - Instantiating training dataset number 1: <pinnsjax.data.DiscreteMeshSampler>
[2023-10-19 07:51:44,364][jax._src.xla_bridge][INFO] - Unable to initialize backend 'rocm': NOT_FOUND: Could not find registered platform with name: "rocm". Available platform names are: CUDA
[2023-10-19 07:51:44,365][jax._src.xla_bridge][INFO] - Unable to initialize backend 'tpu': INTERNAL: Failed to open libtpu.so: libtpu.so: cannot open shared object file: No such file or directory
[2023-10-19 07:51:45,979][pinnsjax.train][INFO] - Instantiating training dataset number 2: <pinnsjax.data.DirichletBoundaryCondition>
[2023-10-19 07:51:46,007][pinnsjax.train][INFO] - Instantiating validation dataset number 1: <pinnsjax.data.DiscreteMeshSampler>
[2023-10-19 07:51:46,031][pinnsjax.train][INFO] - Instantiating prediction dataset number 1: <pinnsjax.data.DiscreteMeshSampler>
[2023-10-19 07:51:46,034][pinnsjax.train][INFO] - Instantiating datamodule <pinnsjax.data.PINNDataModule>
[2023-10-19 07:51:46,036][pinnsjax.train][INFO] - Instantiating neural net <pinnsjax.models.FCN>
[2023-10-19 07:51:47,149][pinnsjax.train][INFO] - Instantiating model <pinnsjax.models.PINNModule>
[2023-10-19 07:51:47,158][pinnsjax.utils.utils][INFO] - Weights are available.
[2023-10-19 07:51:47,427][pinnsjax.train][INFO] - Instantiating trainer <pinnsjax.trainer.Trainer>
[2023-10-19 07:51:47,430][pinnsjax.train][INFO] - Starting training!
[2023-10-19 07:51:47,430][pinnsjax.trainer.trainer][INFO] - Training with Adam Optimizer
[2023-10-19 07:51:47,448][pinnsjax.utils.utils][ERROR] - 
jax.errors.SimplifiedTraceback: For simplicity, JAX has removed its internal frames from the traceback of the following exception. Set JAX_TRACEBACK_FILTERING=off to include these.

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/reza/tmp/pinns-jax/pinnsjax/utils/utils.py", line 72, in wrap
    metric_dict, object_dict = task_func(
  File "/home/reza/tmp/pinns-jax/pinnsjax/train.py", line 138, in train
    trainer.fit(model=model, datamodule=datamodule)
  File "/home/reza/tmp/pinns-jax/pinnsjax/trainer/trainer.py", line 109, in fit
    loss, extra_variables, model.trainable_variables, model.opt_state = model.train_step(model.trainable_variables,
  File "/home/reza/tmp/pinns-jax/pinnsjax/models/pinn_module.py", line 139, in train_step
    loss_pred, grads = jax.value_and_grad(self.model_step, has_aux=True)(params, batch)
  File "/home/reza/tmp/pinns-jax/pinnsjax/models/pinn_module.py", line 130, in model_step
    loss, preds = self.function_mapping[loss_fn_name](params, data, loss, self.functions)
  File "/home/reza/tmp/pinns-jax/pinnsjax/data/sampler/boundary_condition.py", line 120, in loss_fn
    outputs = functions["forward"](params, x, t)
  File "/home/reza/tmp/pinns-jax/pinnsjax/models/pinn_module.py", line 101, in forward
    outputs = self.functional_net(params, *spatial, time = time)
  File "/home/reza/tmp/pinns-jax/pinnsjax/utils/module_fn.py", line 101, in functional_model_1d
    return _execute_model(net, params, [x], time, output_c)
  File "/home/reza/tmp/pinns-jax/pinnsjax/utils/module_fn.py", line 126, in _execute_model
    outputs_dict = net(params, inputs, time)
  File "/home/reza/tmp/pinns-jax/pinnsjax/models/net/neural_net.py", line 73, in __call__
    z = self.forward(params, z)
  File "/home/reza/tmp/pinns-jax/pinnsjax/models/net/neural_net.py", line 57, in forward
    H = jax.nn.tanh(jnp.dot(H, w) + b)
  File "/home/reza/miniconda3/envs/jax/lib/python3.9/site-packages/jax/_src/numpy/lax_numpy.py", line 3122, in dot
    result = lax.dot_general(a, b, dimension_numbers=(contract_dims, batch_dims),
TypeError: dot_general requires contracting dimensions to have the same shape, got (2,) and (1,).
[2023-10-19 07:51:47,450][pinnsjax.utils.utils][INFO] - Output dir: /home/reza/tmp/pinns-jax/examples/burgers_discrete_forward/outputs/07-51-44
Error executing job with overrides: ['trainer.enable_progress_bar=False']
jax.errors.SimplifiedTraceback: For simplicity, JAX has removed its internal frames from the traceback of the following exception. Set JAX_TRACEBACK_FILTERING=off to include these.

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/reza/tmp/pinns-jax/examples/burgers_discrete_forward/train.py", line 50, in main
    metric_dict, _ = pinnsjax.train(
  File "/home/reza/tmp/pinns-jax/pinnsjax/utils/utils.py", line 84, in wrap
    raise ex
  File "/home/reza/tmp/pinns-jax/pinnsjax/utils/utils.py", line 72, in wrap
    metric_dict, object_dict = task_func(
  File "/home/reza/tmp/pinns-jax/pinnsjax/train.py", line 138, in train
    trainer.fit(model=model, datamodule=datamodule)
  File "/home/reza/tmp/pinns-jax/pinnsjax/trainer/trainer.py", line 109, in fit
    loss, extra_variables, model.trainable_variables, model.opt_state = model.train_step(model.trainable_variables,
  File "/home/reza/tmp/pinns-jax/pinnsjax/models/pinn_module.py", line 139, in train_step
    loss_pred, grads = jax.value_and_grad(self.model_step, has_aux=True)(params, batch)
  File "/home/reza/tmp/pinns-jax/pinnsjax/models/pinn_module.py", line 130, in model_step
    loss, preds = self.function_mapping[loss_fn_name](params, data, loss, self.functions)
  File "/home/reza/tmp/pinns-jax/pinnsjax/data/sampler/boundary_condition.py", line 120, in loss_fn
    outputs = functions["forward"](params, x, t)
  File "/home/reza/tmp/pinns-jax/pinnsjax/models/pinn_module.py", line 101, in forward
    outputs = self.functional_net(params, *spatial, time = time)
  File "/home/reza/tmp/pinns-jax/pinnsjax/utils/module_fn.py", line 101, in functional_model_1d
    return _execute_model(net, params, [x], time, output_c)
  File "/home/reza/tmp/pinns-jax/pinnsjax/utils/module_fn.py", line 126, in _execute_model
    outputs_dict = net(params, inputs, time)
  File "/home/reza/tmp/pinns-jax/pinnsjax/models/net/neural_net.py", line 73, in __call__
    z = self.forward(params, z)
  File "/home/reza/tmp/pinns-jax/pinnsjax/models/net/neural_net.py", line 57, in forward
    H = jax.nn.tanh(jnp.dot(H, w) + b)
  File "/home/reza/miniconda3/envs/jax/lib/python3.9/site-packages/jax/_src/numpy/lax_numpy.py", line 3122, in dot
    result = lax.dot_general(a, b, dimension_numbers=(contract_dims, batch_dims),
TypeError: dot_general requires contracting dimensions to have the same shape, got (2,) and (1,).

Set the environment variable HYDRA_FULL_ERROR=1 for a complete stack trace.
